python_modules/libraries/dagster-dg/dagster_dg/config.py

```diff
@@ -52,115 +161,152 @@ class DgConfig:
     use_dg_managed_environment: bool
     require_local_venv: bool
-    component_package: str
-    is_project: bool
-    is_workspace: bool
-
-
-def _normalize_dg_partial_config(raw_dict: Mapping[str, object]) -> DgPartialConfig:
-    config = {**raw_dict}  # copy so we can modify it
-    if "disable_cache" in config and not isinstance(config["disable_cache"], bool):
-        raise DgValidationError("`disable_cache` must be a boolean.")
-    if "cache_dir" in config:
-        if not isinstance(config["cache_dir"], (Path, str)):
-            raise DgValidationError("`cache_dir` must be a string.")
-        elif isinstance(config["cache_dir"], str):
-            config["cache_dir"] = Path(config["cache_dir"])
-    if "verbose" in config and not isinstance(config["verbose"], bool):
-        raise DgValidationError("`verbose` must be a boolean.")
-    if "builtin_component_lib" in config and not isinstance(config["builtin_component_lib"], str):
-        raise DgValidationError("`builtin_component_lib` must be a string.")
-    if "use_dg_managed_environment" in config and not isinstance(
-        config["use_dg_managed_environment"], bool
-    ):
-        raise DgValidationError("`use_dg_managed_environment` must be a boolean.")
-    if "require_local_venv" in config and not isinstance(config["require_local_venv"], bool):
-        raise DgValidationError("`require_local_venv` must be a boolean.")
-    if "component_package" in config and not isinstance(config["component_package"], str):
-        raise DgValidationError("`component_package` must be a string.")
-    if "is_project" in config and not isinstance(config["is_project"], bool):
-        raise DgValidationError("`is_project` must be a boolean.")
-
-    if "is_workspace" in config and not isinstance(config["is_workspace"], bool):
-        raise DgValidationError("`is_workspace` must be a boolean.")
-
-    if unrecognized_keys := [k for k in config.keys() if k not in DgPartialConfig.__annotations__]:
-        raise DgValidationError(f"Unrecognized fields in configuration: {unrecognized_keys}")
-    return cast(DgPartialConfig, config)
 
 
 # ########################
-# ##### CONFIG CLI OPTIONS
+# ##### PROJECT
+# ########################
+
+
+@dataclass
+class DgProjectConfig:
+    root_module: str
+    components_module: Optional[str] = None
+
+    @classmethod
+    def from_raw(cls, raw: "DgRawProjectConfig") -> Self:
+        return cls(
+            root_module=raw["root_module"],
+            components_module=raw.get("components_module", DgProjectConfig.components_module),
+        )
+
+
+class DgRawProjectConfig(TypedDict):
+    root_module: Required[str]
+    components_module: NotRequired[str]
+
+
+# ########################
+# ##### WORKSPACE
+# ########################
+
+
+@dataclass
+class DgWorkspaceConfig:
+    pass
+
+    @classmethod
+    def from_raw(cls, raw: "DgRawWorkspaceConfig") -> Self:
+        return cls()
+
+
+class DgRawWorkspaceConfig(TypedDict):
+    pass
```

Comment:
We are adding an empty `tool.dg.workspace` config section in this PR but upstack the `projects` field gets added.